ui/app/contexts/metametrics.new.js

```diff
@@ -0,0 +1,174 @@
+      const match = matchPath(location.pathname, { path: PATHS_TO_CHECK, exact: true, strict: true })
+      if (match) {
+        const { path, params } = match
+        const name = PATH_NAME_MAP[path]
+        segment.page({
+          [idTrait]: idValue,
+          name,
+          properties: {
+            url: path,
+            hash: location.hash,
+            // We do not want to send addresses or accounts in any events
+            // Some routes include these as params.
+            params: omit(params, ['account', 'address']),
+          },
+        })
+      } else if (location.pathname !== '/confirm-transaction') {
+        // We have more specific pages for each type of transaction confirmation
+        // The user lands on /confirm-transaction first, then is redirected based on
+        // the contents of state.
+        captureMessage(`${location.pathname} would have issued a page track event to segment, but no route match was found`)
+      }
+    }
+  }, [location, metaMetricsId, participateInMetaMetrics])
+
+  /**
+   * track a metametrics event using segment
+   * e.g metricsEvent({ event: 'Unlocked MetaMask', category: 'Navigation' })
+   *
+   * @param {object}   config - configuration object for the event to track
+   * @param {string}   config.event - event name to track
+   * @param {string}   config.category - category to associate event to
+   * @param {boolean}  [config.isOptIn] - happened during opt in/out workflow
+   * @param {object}   [config.properties] - object of custom values to track, snake_case
+   * @return {undefined}
+   */
+  const trackEvent = useCallback(
+    (config = {}) => {
+      console.log(config)
+      const { event, category, isOptIn = false, properties = {} } = config
+      if (!event) {
+        // Event name is required for tracking an event
+        throw new Error('MetaMetrics trackEvent function must be provided a payload with an "event" key')
+      }
+      if (!category) {
+        // Category must be supplied for every tracking event
+        throw new Error('MetaMetrics events must be provided a category')
+      }
+
+      let excludeMetaMetricsId = config.excludeMetaMetricsId ?? false
+
+      // This is carried over from the old implementation, and will likely need
+      // to be updated to work with the new tracking plan. I think we should use
+      // a config setting for this instead of trying to match the event name
+      const isSendFlow = Boolean(event.match(/^send|^confirm/u))
+      if (isSendFlow && !sendCountIsTrackable(metaMetricsSendCount + 1)) {
+        excludeMetaMetricsId = true
+      }
+      const idTrait = excludeMetaMetricsId ? 'anonymousId' : 'userId'
+      const idValue = excludeMetaMetricsId ? METAMETRICS_ANONYMOUS_ID : metaMetricsId
+
+      if (participateInMetaMetrics || isOptIn) {
+        segment.track({
+          [idTrait]: idValue,
+          event,
+          properties: {
+            ...properties,
```

Comment:
Nit: We should consider disallowing the following property names from being passed in via this "properties" parameter: `revenue`, `currency`, and `value`. These are [reserved property names](https://segment.com/docs/connections/spec/track/#properties) and are assumed to have a specific meaning by Segment. `value`especially seems easy to accidentally use for some other purpose.

Passing these reserved names in via distinct config properties seems safer? At least then we can add a JSDoc description explaining what to use it for.

Reply:
I did exactly this. Check the JSDoc to make sure my language is ok.

Reply:
Looks good to me!